### PR TITLE
Add deterministic test factories (Faker + Hypothesis)

### LINKED ISSUE #146: Implement test factories for common test objects with faker and hypothesis support
## Overview

Create a unified test factory system to generate common test objects (likes, comments, follows, posts, agents, etc.) with a consistent interface. This will reduce boilerplate, improve test maintainability, and add robustness through faker and hypothesis integration.

## Problem

Currently, tests manually instantiate objects with repetitive boilerplate across many test files. This leads to:
- Inconsistent test data patterns
- Difficult maintenance when model signatures change
- Lack of property-based testing capabilities
- High cognitive load for writing new tests

## Current State - Manual Instantiation Counts

Search results show the following manual instantiations across test files:

| Class | Count | Primary Usage |
|-------|-------|---------------|
| `BlueskyFeedPost` | 49 | Feed/post generation in tests |
| `RunConfig` | 46 | Run configuration setup |
| `GenerationMetadata` | 35 | Action metadata creation |
| `SocialMediaAgent` | 20 | Agent creation |
| `GeneratedLike` | 4 | Like action generation |
| `GeneratedComment` | 4 | Comment action generation |
| `GeneratedFollow` | 5 | Follow action generation |
| `Like` | 4 | Core like action |
| `Comment` | 4 | Core comment action |
| `Follow` | 5 | Core follow action |

## Proposed Solution

Implement test factories with:

### 1. Factory Pattern
- Create `tests/factories/` module with factory classes for each model
- Support builder pattern for customization
- Provide sensible defaults for all fields
- Support both single object and batch creation

### 2. Faker Integration
- Use `faker` library for realistic test data (names, handles, timestamps, text)
- Configurable locales and seeds for reproducibility
- Domain-specific providers (e.g., bluesky handles, post URIs)

### 3. Hypothesis Integration
- Provide hypothesis strategies for property-based testing
- Enable generation of edge cases and boundary conditions
- Support for testing invariants across random inputs

### 4. API Design Example

```python
# Simple usage
post = PostFactory.create()
agent = AgentFactory.create(handle="custom.bsky.social")
config = RunConfigFactory.create(num_agents=5)

# Batch creation
posts = PostFactory.create_batch(10)

# Custom overrides
like = GeneratedLikeFactory.create(
    agent_handle="agent1.bsky.social",
    post_id="post123"
)

# With faker
bio = GeneratedBioFactory.create()  # Uses faker for realistic bio text

# Hypothesis strategies
from tests.factories.strategies import bluesky_post_strategy
@given(post=bluesky_post_strategy())
def test_property(post):
    assert validate_post(post)
```

### Phase 1: Core Infrastructure
- [ ] Create `tests/factories/` module structure
- [ ] Add `faker` and `hypothesis` to dev dependencies
- [ ] Implement base factory class with common utilities
- [ ] Create factory for `BlueskyFeedPost` (highest count)

### Phase 2: Action & Agent Factories
- [ ] Factory for `GenerationMetadata`
- [ ] Factory for `Like`, `Comment`, `Follow`
- [ ] Factory for `GeneratedLike`, `GeneratedComment`, `GeneratedFollow`
- [ ] Factory for `SocialMediaAgent`

### Phase 3: Configuration Factories
- [ ] Factory for `RunConfig`
- [ ] Factory for `PersistedLike`, `PersistedComment`, `PersistedFollow`
- [ ] Other configuration/model factories as needed

### Phase 4: Migration & Hypothesis
- [ ] Audit all test files for manual instantiations (using rg patterns from analysis)
- [ ] Replace manual instantiations with factory calls
- [ ] Implement hypothesis strategies for property-based tests
- [ ] Add example property-based tests demonstrating usage

### Phase 5: Documentation
- [ ] Write factory usage guide in `tests/README.md`
- [ ] Document faker customization patterns
- [ ] Provide hypothesis strategy examples
- [ ] Update contributing guidelines

## Files with Manual Instantiations

Based on the search, these test files contain manual instantiations:

**High Priority (most instantiations):**
- `tests/db/repositories/test_feed_post_repository.py` (BlueskyFeedPost: ~20)
- `tests/db/repositories/test_feed_post_repository_integration.py` (BlueskyFeedPost: ~15)
- `tests/db/repositories/test_run_repository_integration.py` (RunConfig: ~30)
- `tests/db/repositories/test_generated_bio_repository*.py` (GenerationMetadata: ~25)

**Medium Priority:**
- `tests/simulation/core/test_command_service.py`
- `tests/simulation/core/test_query_service.py`
- `tests/feeds/test_feed_generator.py`
- `tests/db/repositories/test_action_repositories_integration.py`

**Lower Priority:**
- `tests/simulation/core/test_agent_action_rules_validator.py`
- `tests/simulation/core/test_social_media_agent.py`
- Various action generator policy tests

## Search Patterns for Migration

Use these patterns to find all instances during migration:

```bash
# Find all instantiations
rg --type=py -g 'tests/**/*.py' 'GeneratedLike\(' 
rg --type=py -g 'tests/**/*.py' 'GeneratedComment\('
rg --type=py -g 'tests/**/*.py' 'GeneratedFollow\('
rg --type=py -g 'tests/**/*.py' '\bLike\('
rg --type=py -g 'tests/**/*.py' '\bComment\('
rg --type=py -g 'tests/**/*.py' '\bFollow\('
rg --type=py -g 'tests/**/*.py' 'BlueskyFeedPost\('
rg --type=py -g 'tests/**/*.py' 'SocialMediaAgent\('
rg --type=py -g 'tests/**/*.py' 'GenerationMetadata\('
rg --type=py -g 'tests/**/*.py' 'RunConfig\('
```

## Benefits

1. **Reduced Boilerplate**: Less repetitive code in tests
2. **Consistency**: Uniform test data patterns
3. **Maintainability**: Single point of change when models evolve
4. **Robustness**: Faker provides realistic edge cases
5. **Property Testing**: Hypothesis enables comprehensive testing
6. **Readability**: Tests focus on behavior, not data construction

## References

- **Backlink**: PR #135 - https://github.com/METResearchGroup/social_agent_simulation_platform/pull/135
- **Comment**: https://github.com/METResearchGroup/social_agent_simulation_platform/pull/135#discussion_r2854064921
- **Requester**: @mark-torres10

## Related Libraries

- [factory_boy](https://factoryboy.readthedocs.io/): Python factory pattern for tests
- [faker](https://faker.readthedocs.io/): Generate fake data
- [hypothesis](https://hypothesis.readthedocs.io/): Property-based testing

### PR DESCRIPTION
**Pull request by mark-torres10**
## Overview
Add a unified, deterministic test factory system for building simulation domain models in tests. This removes repeated inline Pydantic model construction, improves readability, and enables property-based testing.

## Problem / motivation
Tests frequently constructed domain models inline (posts, runs, turns, metrics, agents, generated/persisted actions, feeds), leading to duplicated boilerplate, inconsistent defaults, and noisy diffs when models change. Issue #146 requests a consistent factory API plus Faker + Hypothesis support.

## Solution
Introduce `tests/factories/` as a test-only factory package backed by a per-test seeded `faker.Faker()` (ContextVar + autouse fixture), plus Hypothesis strategies. Migrate test callsites to use factories (including `GeneratedFeed`, `BlueskyFeedPost`, and repository/API models) and replace repeated mock-engine helpers with a single `EngineFactory`.

## Happy Flow
1. **Per-test Faker** – `tests/conftest.py` seeds a `faker.Faker()` instance from `request.node.nodeid` and stores it in a `ContextVar` (`tests/factories/context.py`).
2. **Factories** – tests call `tests.factories.*Factory.create(...)` / `create_batch(...)` which pulls the per-test Faker via `tests/factories/context.py:get_faker()` and returns validated domain models.
3. **Property tests** – `tests/factories/strategies.py` provides Hypothesis strategies used by `tests/test_factories_hypothesis.py`.
4. **API/run tests** – `tests/factories/engine.py:EngineFactory.create_completed_run_engine(...)` provides a consistent mock engine shape for API/service tests.

## Data Flow
- `pytest` → autouse `fake` fixture (`tests/conftest.py`) → `tests/factories/context.py` ContextVar → factory `create()` methods (`tests/factories/*.py`) → constructed Pydantic models returned to the test.

## Changes
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/pyproject.toml`: add `faker` + `hypothesis` to test extras.
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/tests/conftest.py`: autouse per-test seeded Faker fixture wired into factories context.
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/tests/factories/`: factory package (`BaseFactory`, `context`, helpers, model factories, Hypothesis strategies).
  - Adds `GeneratedFeedFactory`, `BlueskyProfileFactory`, record factories (`AgentRecordFactory`, `AgentBioFactory`, `UserAgentProfileMetadataFactory`), and `TurnResultFactory`.
  - Adds `EngineFactory` to replace `_make_mock_engine_*` helpers.
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/tests/*`: migrate tests to use factories instead of inline model construction.
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/docs/plans/2026-02-25_test_factories_146146/`: model inventory + before/after constructor counts.
- `/Users/mark/.codex/worktrees/f6cc/agent_simulation_platform/docs/runbooks/CREATE_NEW_PYTHON_TESTS.md`: runbook for writing new pytest tests using factories + `uv run pytest`.

## Manual Verification
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv sync --extra test`
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv run pytest` — `593 passed, 2 skipped`
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv run ruff check .`
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv run ruff format .`
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv run pyright .`
- `cd /Users/mark/.codex/worktrees/f6cc/agent_simulation_platform && uv run pre-commit run --all-files`

Closes #146.

## Summary by CodeRabbit

* **Chores**
  * Added a comprehensive test-factory system and an autouse per-test deterministic Faker fixture.
  * Added faker and hypothesis to test dependencies; updated .gitignore to ignore Hypothesis artifacts.

* **Tests**
  * Migrated many tests to factory-based creation; added factory smoke tests and Hypothesis property-based tests.
  * Introduced numerous factory modules and helpers to centralize test object creation.

* **Documentation**
  * Added model-inventory and constructor-count reference documents.

### CODE REVIEW

File: tests/api/test_simulation_run.py [hunk 1]
```diff
@@ -6,9 +6,14 @@
...
+from tests.factories import (
+    RunConfigFactory,
+    RunFactory,
+    RunMetricsFactory,
+    TurnMetadataFactory,
+    TurnMetricsFactory,
+)
 
 
 def _make_mock_engine_for_completed_run(
```
Comment (mark-torres10) [inline tests/api/test_simulation_run.py]: can we replace this with a engine factory?
